src/FakeItEasy.Analyzer/SymbolHelpers.cs

```diff
@@ -21,20 +21,38 @@ internal static class SymbolHelpers
         /// <returns>The symbol for the invocation call.</returns>
         internal static IMethodSymbol GetCalledMethodSymbol(InvocationExpressionSyntax call, SyntaxNodeAnalysisContext context)
         {
-            var name = (call?.Expression as MemberAccessExpressionSyntax)?.Name
-                ?? call?.Expression as IdentifierNameSyntax;
-
-            var method =
-                name == null
-                    ? null
-                    : context.SemanticModel.GetSymbolInfo(name).Symbol as IMethodSymbol;
+            var method = context.SemanticModel.GetSymbolInfo(call).Symbol as IMethodSymbol;

```

Comment:
This change and the one in `GetAccessedPropertySymbol` are unrelated to this PR, but I realized while writing `GetAccessedIndexerSymbol` that they could be simplified. The changes are in a separate commit, so I can easily move them to a separate PR if you prefer.